core/state/statedb.go

```diff
@@ -630,6 +666,13 @@ func (s *StateDB) createObject(addr common.Address) (newobj, prev *stateObject)
 	newobj = newObject(s, addr, nil)
 	if prev == nil {
 		s.journal.append(createObjectChange{account: &addr})
+		if s.logger != nil {
+			// Precompiled contracts are touched during a call.
+			// Make sure we avoid emitting a new account event for them.
+			if _, ok := s.precompiles[addr]; !ok {
+				s.logger.OnNewAccount(addr)
+			}
```

Comment:
I understand the concern. For the tracers it will be definitely weird to receive NewAccount events for precompiles. Where do you suggest we hide their creation? This is the cleanest solution I could find.
The stateDB is already aware of precompiles in a sense to count them in the access list.

Reply:
> Where do you suggest we hide their creation? 

I don't understand why you see a need to hide this. It is technically accurate to trigger a create-event for it. 

It happens a total of 5 (? I think, because there were 5 originally, and the ones that were added later were 'created' in advance before they became precompiles) times throughout the chain history. 

>  The stateDB is already aware of precompiles in a sense to count them in the access list.

Well, no. The access list contains addresses, it doesn't mean they 'exist'. An accesslist may for example contain addresses of contracts-to-be-deployed and slots-to-be-created. 

When I said "the precompiles does not exist in the state" [sic], I meant it literally: they do not leave a mark on the stateroot. They are not in the trie. Until the account is created, at some point.

Reply:
> From a trie perspective, new accounts are made in the precompile-range whenever someone sends funds to it.

I failed to understand your point at first. Given that precompiles really are created in the state it's wrong not to emit an event for them. That said on a testnet it's possible the tracer will receive a bunch of these events for precompiles because we touch them in statedb when they are called. The tracer will have to handle that case. AFAIK there is not really a way to detect whether a precompile was only touched because it was called or it will be created in the state (a future balance update is coming).

That brings me to the second point: OnNewAccount probably needs a counterpart OnAccountDeleted, to help someone tracking changes to the state to know when an empty account was cleared.

Or alternatively drop OnNewAccount on the ground that it's better to track (balance, nonce, code, storage) changes anyway.

Reply:
> That said on a testnet it's possible the tracer will receive a bunch of these events for precompiles because we touch them in statedb when they are called.

It's pretty common to put `1` wei on all low addresses, in the genesis, just to avoid complexities regarding precompiles and emptiness. So I don't think this will be much of an issue. 

> That brings me to the second point: OnNewAccount probably needs a counterpart OnAccountDeleted, to help someone tracking changes to the state to know when an empty account was cleared.

Myeah, maybe so. But does that really ever happy anymore? It can't right?

Reply:
> Or alternatively drop OnNewAccount on the ground that it's better to track (balance, nonce, code, storage) changes anyway.

Actually, that sounds like a pretty neat idea. It's what 'consumers' care about, isn't it, rather than exactly what the trie internals are at any given point ?